crates/db_views/src/comment_view.rs

```diff
@@ -315,6 +316,25 @@ impl CommentView {
   }
 }
 
+impl ToSlimView for CommentView {
+  type SlimView = CommentSlimView;
+  fn map_to_slim(self) -> Self::SlimView {
+    Self::SlimView {
+      comment: self.comment,
+      creator: self.creator,
+      counts: self.counts,
+      creator_banned_from_community: self.creator_banned_from_community,
+      banned_from_community: self.banned_from_community,
+      creator_is_moderator: self.creator_is_moderator,
+      creator_is_admin: self.creator_is_admin,
+      subscribed: self.subscribed,
+      saved: self.saved,
+      creator_blocked: self.creator_blocked,
+      my_vote: self.my_vote,
+    }
+  }
+}
```

Comment:
Mapping to a slimmer type seems like the easiest way to go. 

The much harder way would be to construct an entire new `comment_slim_view.rs`, and try to abstract a lot of the logic in this file. This isn't worth it IMO, because the comment_view still needs to join to the post table anyway to do filtering.

Reply:
Is there much benefit to this at all then? It would still use unnecessary bandwidth to the database (which is uncompressed afaik). API responses will be smaller but they are compressed anyway, so its probably no real difference. 

Changing comment_view.rs to take a `slim` parameter shouldnt be so hard, you need to change the select clause and the return type. Otherwise I would rather leave it unchanged and close the issue as wontfix.

Reply:
I think so, because the main issue isn't really database join speed, but the amount of data. Even if its compressed, if you are fetching 300 comments, it currently includes the post and community for every single comment. That means like 40+ fields, including things like the community sidebar and post description, for every comment.

Reply:
```sh
curl "https://lemmy.ml/api/v3/comment/list?post_id=3192572&max_depth=8&sort=Top&type_=All&saved_only=false" -w '%{size_download}'  -so /dev/null
1053421
curl "https://lemmy.ml/api/v3/comment/list?post_id=3192572&max_depth=8&sort=Top&type_=All&saved_only=false" -w '%{size_download}'  -so /dev/null --compressed
82554
```

So only 80kb compressed data for 300 comments. I suppose each of these would be decreased by half with this change. On the other hand the db connection has no compression so it still needs to transfer the full 1mb.

Reply:
I'll take a crack at it... it'd be trivial to do with copy paste, but possibly really difficult to abstract it in a smart way, because diesel's boxed query return types are a nightmare to deal with.

I'd essentially keep all the joins the same, but change the `.select(...` ed columns.